File: app/src/main/java/chess/android/arduino/telescope/coordinates/HourAngle.java
package chess.android.arduino.telescope.coordinates;

public class HourAngle  extends Coordinate{


    public HourAngle(int hour, int minute, int second) {
        super(hour, minute, second);
    }

    @Override
    public double toDegrees() {
        return (((this.getHour()*3600.0d)+
                (this.getMinute()*60.0d)
                +(this.getSecond()*1.0d))*15.0d)
                /3600.0d;
    }
}

File: app/src/main/java/chess/android/arduino/telescope/coordinates/HourAngle.kt
package chess.android.arduino.telescope.coordinates

class HourAngle(hour: Int, minute: Int, second: Int) : Coordinate(hour, minute, second) {
    override fun toDegrees(): Double {
        return ((hour * 3600.0 +
                minute * 60.0
                + second * 1.0) * 15.0
                / 3600.0)
    }
}